### ISSUE TITLE
Changing the value of a nullable value type with the VS2015 debugger fails

### ISSUE THREAD

**Issue by eamodio** [dotnet/roslyn#3211]
Hi,

Not sure if this is really the right place to report this, but I'm guessing it might be - if not please direct me to where ever it should be reported.

I have included a super-simple C# program which can be used to exhibit the behavior.

```c#
public class Program
{
    public static void Main()
    {
        var foo = new Foo();

        Debugger.Break();

        // At this point add foo to the watch
        // Try to change foo.IntValue to an int (something other than 1) and it will get set to 1
        // Try to change foo.BoolValue to false and it will get set to true

        // The same issue occurs using the immediate window to set the value or the hover info-tip
    }

    public class Foo
    {
        public int? IntValue { get; set; }
        public bool? BoolValue { get; set; }
    }
}
```
Can download the project here: https://www.dropbox.com/s/9bhj4g2ddhcrr3w/VS2015-debug.zip?dl=0

To reproduce the issue, run the program and at the break, using the watch or immediate, try changing the value of either of the Foo properties. Neither will get set to the correct value.

Please let me know if I can provide any more details or assistance.

Thanks!

**Comment by tmat** [comment]
@eamodio Yes, this is the right place to report these issues. Thanks for doing so. I can confirm that the current behavior is incorrect.

```
foo.IntValue
null
foo.IntValue = 10
0x0000000a
foo.IntValue
0x00000001
foo.IntValue
0x00000001
foo.IntValue = 100
0x00000064
foo.IntValue
0x00000001
```

**Comment by cston** [comment]
Opened internal bug 1180851.

**Comment by eamodio** [comment]
@ManishJayaswal curiously what does the Blocked label mean? (Or is there a legend/descriptions of the labels somewhere?)

**Comment by ManishJayaswal** [comment]
@eamodio We need a fix for this on the debugger side ( which is not open source) and we have an internal bug tracking that. Once that is resolved we will update this issue.
We use the blocked tag for our bookkeeping indicating that it is not actionable yet.

**Comment by pnelsonmsft** [comment]
The fix is currently targeted for Update 1

**Comment by eamodio** [comment]
@ManishJayaswal ah - awesome. Thank you!
@pnelsonmsft Sweet! :)

**Comment by pnelsonmsft** [comment]
I fixed this bug in the debugger code base.  PR#2549

**Comment by ManishJayaswal** [comment]
@pnelsonmsft thanks!
So this should be available in update1 release of VS2015?

**Comment by pnelsonmsft** [comment]
Yes, the fix should make Update 1.

**Comment by ManishJayaswal** [comment]
Great .. I will close this issue here then.
